localstack/utils/docker.py

```diff
@@ -596,5 +803,559 @@ def mountable_tmp_file():
+                image_names = list(map(lambda image_name: image_name.split(":")[0], image_names))
+            if strip_latest:
+                Util.append_without_latest(image_names)
+            return image_names
+        except APIError:
+            raise ContainerException()
+
+    def get_container_logs(self, container_name_or_id: str, safe=False) -> str:
+        try:
+            container = self.client().containers.get(container_name_or_id)
+            return to_str(container.logs())
+        except NotFound:
+            if safe:
+                return ""
+            raise NoSuchContainer(container_name_or_id)
+        except APIError:
+            if safe:
+                return ""
+            raise ContainerException()
+
+    def inspect_container(self, container_name_or_id: str) -> Dict[str, Union[Dict, str]]:
+        try:
+            return self.client().containers.get(container_name_or_id).attrs
+        except NotFound:
+            raise NoSuchContainer(container_name_or_id)
+        except APIError:
+            raise ContainerException()
+
+    def inspect_image(self, image_name: str) -> Dict[str, Union[Dict, str]]:
+        try:
+            return self.client().images.get(image_name).attrs
+        except NotFound:
+            raise NoSuchImage(image_name)
+        except APIError:
+            raise ContainerException()
+
+    def get_container_ip(self, container_name_or_id: str) -> str:
+        return self.inspect_container(container_name_or_id)["NetworkSettings"]["IPAddress"]
+
+    def has_docker(self) -> bool:
+        try:
+            if not self.docker_client:
+                return False
+            self.client().ping()
+            return True
+        except APIError:
+            return False
+
+    def start_container(
+        self,
+        container_name_or_id: str,
+        stdin=None,
+        interactive: bool = False,
+        attach: bool = False,
+        flags: Optional[str] = None,
+    ) -> Tuple[bytes, bytes]:
+        LOG.debug("Starting container %s", container_name_or_id)
+        try:
+            container = self.client().containers.get(container_name_or_id)
+            stdout = to_bytes(container_name_or_id)
+            stderr = b""
+            if interactive or attach:
+                params = {"stdout": 1, "stderr": 1, "stream": 1}
+                if interactive:
+                    params["stdin"] = 1
+                sock = container.attach_socket(params=params)
+                sock = sock._sock if hasattr(sock, "_sock") else sock
+                container.start()
+                with sock:
+                    try:
+                        if stdin:
+                            sock.sendall(to_bytes(stdin))
+                            sock.shutdown(socket.SHUT_WR)
+                        stdout, stderr = self._read_from_sock(sock, False)
+                    except socket.timeout:
+                        LOG.debug("timeout")
+                        pass
```

Comment:
Let's better add a slightly more informative log message:

```
                    except socket.timeout:
                        LOG.debug(f"Socket timeout when talking to the I/O streams of Docker container '{container_name_or_id}'")
```
(also, `pass` could be removed)